packages/remix-react/links.ts

```diff
@@ -254,9 +254,14 @@ export async function prefetchStyleLinks(
     }
   }
 
-  // don't block for non-matching media queries
+  // don't block for non-matching media queries, or for stylesheets that are
+  // already in the DOM (active route revalidations)
   let matchingLinks = styleLinks.filter(
-    (link) => !link.media || window.matchMedia(link.media).matches
+    (link) =>
+      (!link.media || window.matchMedia(link.media).matches) &&
+      document.querySelector(
+        `head link[rel="stylesheet"][href="${link.href}"]`
```

Comment:
Do we care about the head specifically? If a user manually renders a link tag in the root component, but in the body for some reason, should that not also be deduped?

Reply:
nah - we can remove it in case they do any manual rendering of their own or the user puts `<Links>` into `body` for some reason